### ISSUE TITLE
Datagrid regression from 0.9.2.2 to 0.9.2.3 onwards

### ISSUE THREAD

**Issue by vbray1979** [stsrki/Blazorise#1609]
OnReadData is called twice with the first call not containing "filter"/"sorting" data.
```
<DataGrid TItem="PortfolioModel"
                  Data="@PortfolioList"
                  ReadData="@OnReadData"
                  PageSize="25"
                  TotalItems="@TotalPortfolio"
                  Filterable="true"
                  ShowPager="true"
                  Sortable="true"
                  Bordered="true"
                  Striped="true"
                  Hoverable="false"
                  Narrow="true"
                  Responsive="true"
                  RowSelectable=@((item)=>false)> 
... 
<DataGridColumn TItem="PortfolioModel" Field="PortfolioCode" Caption="Portfolio code" Sortable="true" Direction="SortDirection.Ascending">
</DataGrid>
```
OnReadData is called twice. In the first call, DataGridReadDataEventArgs do not contain "filter"/"sorting" data. (i.e. PortfolioCode sorting ascending)
`async Task OnReadData(DataGridReadDataEventArgs<PortfolioModel> model)`

This is working fine in version 0.9.2.2

**Comment by paulosdfigueiredo** [comment]
This is related with commit c68bcd8d9d00b66cc2fa1ccb3b75445be3eb89b4. 

The first call is triggered by SubscribeOnPageSizeChanged before all the columns are hooked.

**Comment by stsrki** [comment]
I'm trying to reproduce this error but for me, it works. The `ReadData` is fired only once and all the filter and sorting data are present in eventArgs.

Any chance you guys can post a full sample page to reproduce it?

**Comment by vbray1979** [comment]
Hi Mladen,

This is replicable using the following code and navigating back and forth to the view. "OnReadData" get called twice and in the first call "OrderBy" is empty.

```
public class PortfolioModel
{
   public string PortfolioCode { get; set; }
}
```

```
<DataGrid TItem="PortfolioModel"
          Data="@PortfolioList"
          ReadData="@OnReadData"
          PageSize="25"
          TotalItems="@TotalPortfolio"
          Filterable="true"
          ShowPager="true"
          Sortable="true"
          Bordered="true"
          Striped="true"
          Hoverable="false"
          Narrow="true"
          Responsive="true"
          RowSelectable=@((item)=>false)>
    <DataGridCommandColumn Width="50px" TItem="PortfolioModel" />
    <DataGridColumn TItem="PortfolioModel" Field="PortfolioCode" Caption="Portfolio code" Sortable="true" Direction="SortDirection.Ascending">

    </DataGridColumn>
</DataGrid>

@code{
    int TotalPortfolio;
    PortfolioModel[] PortfolioList;

    async Task OnReadData(DataGridReadDataEventArgs<PortfolioModel> model)
    {
        var OrderBy = model.Columns.Where(x => x.Direction == SortDirection.Ascending).Select(x => x.Field.Replace('.', '/')).ToArray();
        var OrderByDesc = model.Columns.Where(x => x.Direction == SortDirection.Descending).Select(x => x.Field.Replace('.', '/')).ToArray();

        PortfolioList = new List<PortfolioModel>()
        {
            new PortfolioModel(){PortfolioCode = "A"},
            new PortfolioModel(){PortfolioCode = "B"},
            new PortfolioModel(){PortfolioCode = "C"},
            new PortfolioModel(){PortfolioCode = "D"}
        }.ToArray();

        TotalPortfolio = 1;
    }
}
```

**Comment by paulosdfigueiredo** [comment]
Looks like the problem is related with parameter PageSize, probably triggers PageSizeChanged too soon.

**Comment by stsrki** [comment]
Fixed for 0.9.2.x. so I'm closing it.

For 0.9.3 I just need to merge @paulosdfigueiredo PR
